packages/frontend/src/components/UserSettingsModal/AppearancePanel/index.tsx

```diff
@@ -0,0 +1,78 @@
+import { Button, Intent } from '@blueprintjs/core';
+import { ECHARTS_DEFAULT_COLORS } from 'common';
+import React, { FC, useEffect, useState } from 'react';
+import { useApp } from '../../../providers/AppProvider';
+import { useTracking } from '../../../providers/TrackingProvider';
+import SeriesColorPicker from '../../ChartConfigPanel/Series/SeriesColorPicker';
+import {
+    AppearanceColorWrapper,
+    AppearancePanelWrapper,
+    ColorLabel,
+    ColorPalette,
+    Title,
+} from './AppearancePanel.styles';
+
+const AppearanceColor: FC<{
+    color: string;
+    index: number;
+    onChange: (value: string) => void;
+}> = ({ color, index, onChange }) => {
+    return (
+        <AppearanceColorWrapper>
+            <p>Color {index + 1}</p>
+            <div style={{ display: 'flex' }}>
+                <SeriesColorPicker color={color} onChange={onChange} />
+                <ColorLabel>{color}</ColorLabel>
+            </div>
+        </AppearanceColorWrapper>
+    );
+};
+const AppearancePanel: FC = () => {
+    const { track } = useTracking();
+    const { showToastSuccess } = useApp();
+
+    let [colors, setColors] = useState<string[]>(
+        ECHARTS_DEFAULT_COLORS.slice(0, 8),
+    ); //[...Array(8).keys()];
+    console.log('colors', colors);
+
+    // FIXME This should trigger a render if colors change ???
+    useEffect(() => {
+        setColors(colors);
+    }, [colors]);
+
+    //TODO add form
+    return (
+        <AppearancePanelWrapper>
+            <Title>Default chart colours</Title>
+            <ColorPalette>
+                {colors.map((color, index) => (
+                    <AppearanceColor
+                        color={color}
+                        index={index}
+                        onChange={(colorChange) => {
+                            colors[index] = colorChange;
+                            setColors(colors);
```

Comment:
Here you will need to set a new array to the state. `setColors([...colors])`, this happens because setState/useState only re-renders if the previews state has changed, but since arrays in Js are reference types, when you change an item within the array in Js's eyes this two arrays are the same, even if the content within this array is now different.

By doing this you can also get rid of the use state, as the `setState` will now manage the re render

[this discussion](https://stackoverflow.com/questions/56266575/why-is-usestate-not-triggering-re-render) can give you more context